src/lib/middleware/pageCacheMiddleware.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  // @ts-ignore
+  PAGE_CACHE_ENABLED,
+  // @ts-ignore
+  PAGE_CACHE_TYPES,
+  // @ts-ignore
+  PAGE_CACHE_NAMESPACE,
+  // @ts-ignore
+  PAGE_CACHE_VERSION,
+  // @ts-ignore
+  PAGE_CACHE_TIMEOUT_MINUTES,
+} from "../../config"
+import { Request, Response, NextFunction } from "express"
+const cache = require("lib/cache.coffee")
+const cacheablePageTypes: string[] = PAGE_CACHE_TYPES.split("|")
+
+// Middleware will `next` and do nothing if any of the following is true:
+//
+// * page cache feature is disabled.
+// * a user is signed in.
+// * this isnt a supported cacheable path (there is an allow-list set in ENV).
+//
+// At this point, the HTML is looked up in the cache.
+// If available, the request can be responded to and ended immediately.
+// Otherwise, `next` is called and the request continues in our app.
+export const pageCacheMiddleware = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!PAGE_CACHE_ENABLED) return next()
+
+  const pageType = req.path.split("/")[1]
```

Comment:
If you could point to that, I'd be happy to see? Unless we actively want to move homepage or partner profile ('special' ones, not the artist/collect/collection ones done here), we could also just leave as-is and follow up with that.

Reply:
Looks like it's actually 2 client-side locations:

https://github.com/artsy/force/blob/b8a3d849b08ce4c6a218716ce844fd2e63bd6183/src/mobile/analytics/before_ready.js#L13

https://github.com/artsy/force/blob/909916fd1bcae597901fa159ed1645a521e3fa54/src/desktop/analytics/main_layout.js#L10

And then `res.locals.sd.PAGE_TYPE` is explicitly set for those special pages, to `home` for `/`, `partner` for partner profiles, `fair` for fair profiles, etc.

It's probably not reusable as-is, but it might be nice to share the logic somehow.

Reply:
Cool, yea, it's just a `split("/")` on a path (or URL), so I think this is fine for now. I thought there was some code in volley that had a predefined map of url's to types, or something like that.

Reply:
"Unresolving" this just to say: there's no map of URLs to types, but there are lots of custom updates to `res.locals.sd.PAGE_TYPE` [sprinkled around the code](https://github.com/artsy/force/search?q=page_type&unscoped_q=page_type). _If_ we could reuse that value here instead, this caching could support those pages too. I'm just not sure if that's even possible from this middleware.

Reply:
I don't think those are reusable. The `window.location` ones are client-side, whereas this middleware location is server-side.

The special ones which set a value in `res.locals` can definitely continue to be used (and for others, this fallback, `req.path.split("/")` is sufficient). I think doing a simple parsing of `req.path` is pretty idiomatic tbh.